File: DevExpress-Demo/CS/RibbonDemo.Wpf/Helpers/EditWidthConverter.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Windows.Data;
using System.Windows.Markup;
using DevExpress.Xpf.Core;

namespace RibbonDemo
{
    public class EditWidthConverter : MarkupExtension, IValueConverter {
        public double EditWidth { get; set; }
        public double TouchScaleFactor { get; set; }

        public EditWidthConverter() { TouchScaleFactor = 2d; }
        public override object ProvideValue(IServiceProvider serviceProvider) { return this; }

        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            var walker = value as ThemeTreeWalker;
            if (walker != null && (walker.IsTouch || walker.ThemeName == Theme.TouchlineDarkName))
                return EditWidth * TouchScaleFactor;
            return EditWidth;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) { throw new NotImplementedException(); }
    }
}

File: DevExpress-Demo/VB/BarsDemo.Wpf/Helpers/EditWidthConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data
Imports System.Windows.Markup
Imports DevExpress.Xpf.Core

Namespace BarsDemo

    Public Class EditWidthConverter
        Inherits MarkupExtension
        Implements IValueConverter

        Public Property EditWidth As Double

        Public Property TouchScaleFactor As Double

        Public Sub New()
            TouchScaleFactor = 2R
        End Sub

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim walker = TryCast(value, ThemeTreeWalker)
            If walker IsNot Nothing AndAlso (walker.IsTouch OrElse Equals(walker.ThemeName, Theme.TouchlineDarkName)) Then Return EditWidth * TouchScaleFactor
            Return EditWidth
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
